src/components/character-controller.js

```diff
@@ -120,9 +123,23 @@ AFRAME.registerComponent("character-controller", {
         z: root.rotation.z * THREE.Math.RAD2DEG
       });
 
-      this.el.setAttribute("position", root.position);
-
       this.pendingSnapRotationMatrix.identity(); // Revert to identity
+
+      //copied from aframe-extras movement-controls
+      if (this.velocity.lengthSq() > EPS) {
+        let start = root.position;
+        end.copy(this.velocity).multiplyScalar(dt / 1000).add(start);
```

Comment:
Velocity is already applied above, `root.position` is already the target position we are trying to move to.